design-docs/007-basic-expiration.md

```diff
@@ -0,0 +1,561 @@
+# Basic expiration algorithm
+
+## What are _expiration_ and _garbage collection_
+
+Icechunk keeps exhaustive track of history. Currently every snapshot ever written is available
+in object store and never deleted. Many times, that is not what users want. Our experience with
+Arraylake has taught us that 
+users usually want to keep "some history" but not all of it. The reason is:
+wasted storage. If they never go back to check previous versions, why would they want to pay
+for the extra storage.
+
+There are many ways to select what versions (snapshots) should be kept
+around and which ones should be deleted. In our experience, a good balance of
+simplicity and power is: _Expire snapshots older than certain threshold_.
+
+The expiration process has two stages: history edition (aka expiration)
+and garbage collection.
+
+During history edition, references and snapshots can be edited and added
```

Comment:
What does "references" mean here? The manifest files?

Reply:
done